include/mata/nfa/nfa.hh

```diff
@@ -291,38 +258,119 @@ public:
+     */
+    bool empty_language(Run* cex = nullptr) const;
+
+    /**
+     * @brief Test whether an automaton is deterministic.
      *
-     * The value of the already existing symbols will NOT be overwritten.
+     * I.e., whether it has exactly one initial state and every state has at most one outgoing transition over every
+     *  symbol.
+     * Checks the whole automaton, not only the reachable part
      */
-    void add_symbols_to(OnTheFlyAlphabet& target_alphabet) const;
-}; // struct Nfa.
+    bool is_deterministic() const;
 
-/**
-  * Fill @p alphabet with symbols from @p nfa.
-  * @param[in] nfa NFA with symbols to fill @p alphabet with.
-  * @param[out] alphabet Alphabet to be filled with symbols from @p nfa.
-  */
-void fill_alphabet(const Nfa& nfa, mata::OnTheFlyAlphabet& alphabet);
+    /**
+     * @brief Test for automaton completeness with regard to an alphabet.
+     *
+     * An automaton is complete if every reachable state has at least one outgoing transition over every symbol.
+     */
+    bool is_complete(Alphabet const* alphabet = nullptr) const;
+
+    /**
+     * Fill @p alphabet with symbols from @p nfa.
+     * @param[in] nfa NFA with symbols to fill @p alphabet with.
+     * @param[out] alphabet Alphabet to be filled with symbols from @p nfa.
+     */
+    void fill_alphabet(mata::OnTheFlyAlphabet& alphabet) const;
+
+    /// Is the language of the automaton universal?
+    bool is_universal(const Alphabet& alphabet, Run* cex = nullptr,
+                      const ParameterMap& params = {{ "algorithm", "antichains" }}) const;
+    /// Is the language of the automaton universal?
+    bool is_universal(const Alphabet& alphabet, const ParameterMap& params) const;
+
+    /// Checks whether a word is in the language of an automaton.
+    bool is_in_lang(const Run& word) const;
+    /// Checks whether a word is in the language of an automaton.
+    bool is_in_lang(const Word& word) { return is_in_lang(Run{ word, {} }); }
+
+    /// Checks whether the prefix of a string is in the language of an automaton
+    bool is_prfx_in_lang(const Run& word) const;
 
+    std::pair<Run, bool> get_word_for_path(const Run& run) const;
+
+    /**
+     * @brief Make NFA complete in place.
+     *
+     * For each state 0,...,this->num_of_states()-1, add transitions with "missing" symbols from @p alphabet
+     *  (symbols that do not occur on transitions from given state) to @p sink_state. If @p sink_state does not belong
+     *  to the NFA, it is added to it, but only in the case that some transition to @p sink_state was added.
+     * In the case that NFA does not contain any states, this function does nothing.
+     *
+     * @param[in] alphabet Alphabet to use for computing "missing" symbols.
+     * @param[in] sink_state The state into which new transitions are added.
+     * @return True if some new transition was added to the NFA.
+     */
+    bool make_complete(const Alphabet& alphabet, State sink_state);
```

Comment:
just complete? the make anyway looks like making a new automaton

Reply:
We can have `complete()` as the operation and `is_complete()` as the check for completeness.

Reply:
ok

Reply:
Anyone else any opinions?